homeassistant/components/homekit/type_switches.py

```diff
@@ -241,13 +250,23 @@ def set_state(self, value: bool) -> None:
         _LOGGER.debug("%s: Set switch state to %s", self.entity_id, value)
         self.char_in_use.set_value(value)
         params = {ATTR_ENTITY_ID: self.entity_id}
-        service = SERVICE_TURN_ON if value else SERVICE_TURN_OFF
-        self.async_call_service(DOMAIN, service, params)
+
+        service = (
+            (SERVICE_OPEN_VALVE if value else SERVICE_CLOSE_VALVE)
+            if self.domain == VALVE_DOMAIN
+            else (SERVICE_TURN_ON if value else SERVICE_TURN_OFF)
+        )
```

Comment:
Please split multi-line ternaries into if blocks

Reply:
Fixed in ad582b2